src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py

```diff
@@ -804,8 +809,19 @@ def __call__(
             prompt_embeds = torch.cat([negative_prompt_embeds, prompt_embeds])
 
         # 4. Prepare timesteps
-        self.scheduler.set_timesteps(num_inference_steps, device=device)
-        timesteps = self.scheduler.timesteps
+        if timesteps is not None:
```

Comment:
very nice! 

Could we maybe factor this out into a function:

```py
timesteps = retrieve_timesteps(self.scheduler, timesteps)
```

and then use #copied from for all pipelines?